includes/admin_import.php

```diff
@@ -1,15 +1,28 @@
 <?php
 
-//import and settings
+//Import & Export
 
 if (!function_exists('tsml_import_page')) {
 
 	function tsml_import_page()
 	{
-		global $tsml_data_sources, $tsml_programs, $tsml_program, $tsml_nonce, $tsml_sharing, $tsml_slug, $tsml_change_detect;
-
+		global $tsml_data_sources, $tsml_programs, $tsml_program, $tsml_nonce, $tsml_sharing, $tsml_slug, $tsml_detection_test_mode;
+		
 		$error = false;
-		$tsml_data_sources = get_option('tsml_data_sources', array());
```

Comment:
SInce $tsml_data_sources is being set here. Shouldn't the `global $tsml_data_sources` be removed?

Reply:
Hi @anchovie91471, I come from a MS VB/C# background where we almost never used global variables. Come to think of it, even when I was programming in COBOL on the mainframe we pretty much never used Gobals.  I vaguely remember them from Basic A programs back in the 80's, but that came with lot's of problems, especially when debugging and trying to find where values got changed.  I'm more comfortable keeping variables to a local scope.   I'm not sure what the use of Global variables rules are with PHP, but I assume there is a reason for having them declared that way. Feel free to enlighten me as I'm not real sure what you're getting at.

Thx...

Reply:
I agree completely about the use of globals. It gets especially confusing, at times, to know where a value is coming from when it can change literally anywhere. Hotly contested issue in the PHP community.

As far as the code suggestion, after looking again, I realize you are using it exactly how it's meant. You're declaring it, then setting its value. When I made the suggestion, I was thinking from the standpoint of there being no reason to grab a variable from the global scope and then immediately set it, but obviously that's the whole point so that's my bad.